File: algorithms/neetcode/algorithms/algorithms/src/main/java/com/mycompany/trees/ValidateBinarySearchTree.java
package com.mycompany.trees;

public class ValidateBinarySearchTree {
    public static boolean isValidBST(TreeNode root) {
        if(root == null) return true;
        return dfs(root, null, null);
    }

    private static boolean dfs(TreeNode root, Integer min, Integer max) {
        if(root == null) return true;
        if( (min != null && root.getValue() <=min) || max != null && root.getValue() >= max) {
            return false;
        }
        return dfs(root.left, min, root.getValue()) && dfs(root.right, root.getValue(), max);
    }

    public static void main(String... args) {
        TreeNode root = new TreeNode(2);
        root.left = new TreeNode(1);
        root.right = new TreeNode(3);

        System.out.print(isValidBST(root));

    }
}

File: algorithms/neetcode/trees/validate-binary-search-tree.py

from utils import TreeNode

class S:
    def isValidBST(self, root: TreeNode) -> bool:

        def valid(node, left, right):
            if not node:
                return True
            if not (node.val < right and node.val > left):
                return False;

            return (valid(node.left, left, node.val) and
                    valid(node.right, node.val, right))

        return valid(root, float("-inf"), float("inf"))


root = TreeNode(2)
root.left = TreeNode(1)
root.right = TreeNode(3)

result = S().isValidBST(root)
print(result)
